List unecessary env vars for source instrumentations (#1244)
open-telemetry/opentelemetry-dotnet-instrumentation: docs/README.md

OTEL_RESOURCE_ATTRIBUTES=deployment.environment=staging,service.version=1.0.0
```

> If bytecode instrumentations (see [here](config.md#instrumentations)) are not needed,
> then the following environment variables are not required to be set:
> 
> ```env
> COR_ENABLE_PROFILING
> COR_PROFILER
> COR_PROFILER_PATH_32
> COR_PROFILER_PATH_64
> CORECLR_ENABLE_PROFILING
> CORECLR_PROFILER
> CORECLR_PROFILER_PATH
> CORECLR_PROFILER_PATH_32
> CORECLR_PROFILER_PATH_64
> OTEL_DOTNET_AUTO_INTEGRATIONS_FILE
> ```
> 

Reviewer: ```suggestion
If you don't need [bytecode instrumentations](config.md#instrumentations),
you can unset or remove the following environment variables:

```env
COR_ENABLE_PROFILING
COR_PROFILER
COR_PROFILER_PATH_32
COR_PROFILER_PATH_64
CORECLR_ENABLE_PROFILING
CORECLR_PROFILER
CORECLR_PROFILER_PATH
CORECLR_PROFILER_PATH_32
CORECLR_PROFILER_PATH_64
OTEL_DOTNET_AUTO_INTEGRATIONS_FILE
```

```
